### PR TITLE
Prep NavigationPage for removal of InternalsVisibleTo

### PR DESCRIPTION
**Pull request by hartez**
### Description of Change ###

Add the INavigationPageController interface and update renderers to use it.

### CODE REVIEW

File: Xamarin.Forms.Core/INavigationPageController.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Xamarin.Forms.Internals;
+
+namespace Xamarin.Forms
+{
+	public interface INavigationPageController
+	{
+		Stack<Page> StackCopy { get; }
```
Comment (kingces95) [inline Xamarin.Forms.Core/INavigationPageController.cs]: `Copy`?

File: Xamarin.Forms.Core/INavigationPageController.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+		int StackDepth { get; }
```
Comment (kingces95) [inline Xamarin.Forms.Core/INavigationPageController.cs]: Is this the same as `StackCopy.Count`?

**Comment by kingces95** [comment]
:+1: with one question.

File: Xamarin.Forms.Core/INavigationPageController.cs (hunk 2 above)
Comment (hartez) [inline Xamarin.Forms.Core/INavigationPageController.cs]: It is the same, but much faster. `StackCopy` actually constructs a whole new `Stack<Page>` each time it's called.

**Comment by rmarinho** [comment]
👍

**Pull request merged**